File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.swift
import Foundation

func solution(_ sizes:[[Int]]) -> Int {
    var LargeInt : [Int] = []
    var MiniInt : [Int] = []
    
    for i in sizes {
        // if (i[0][0] > i[0][1]) {
        //     LargeInt.append(i[0][0])
        //     MiniInt.append(i[0][1])
        // } else {
        //     LargeInt.append(i[0][1])
        //     MiniInt.append(i[0][0])
        // }
                if (i[0] > i[1]) {
            LargeInt.append(i[0])
            MiniInt.append(i[1])
        } else {
            LargeInt.append(i[1])
            MiniInt.append(i[0])
        }
    }
    
    let result = LargeInt.max()! * MiniInt.max()!
    
    return result
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        
        int longestOfLongs = 0;
        int longestOfShorts = 0;
        
        
        for (int i = 0; i < sizes.length; i++) {
            int w = sizes[i][0];
            int h = sizes[i][1];
            
            if (w>h) {
                longestOfLongs = Math.max(longestOfLongs, w);
                longestOfShorts = Math.max(longestOfShorts, h);
            } else {
                longestOfLongs = Math.max(longestOfLongs, h);
                longestOfShorts = Math.max(longestOfShorts, w);
            }
        }
        
        return longestOfLongs * longestOfShorts;
    }
}